charmcraft/charm_builder.py

```diff
@@ -224,21 +229,23 @@ def handle_dependencies(self):
 
         # virtualenv with other dependencies (if any)
         if self.requirement_paths:
-            _process_run(["pip3", "--version"])
-
-            venvpath = self.buildpath / VENV_DIRNAME
-            cmd = [
-                "pip3",
-                "install",  # base command
-                "--target={}".format(venvpath),  # put all the resulting files in that specific dir
-            ]
+            staging_venv_dir = self.charmdir / STAGING_VENV_DIRNAME
+            _process_run(["python3", "-m", "venv", str(staging_venv_dir)])
+            pip_cmd = str(staging_venv_dir / "bin" / "pip3")
+
+            _process_run([pip_cmd, "--version"])
+
+            cmd = [pip_cmd, "install", "--no-binary", ":all:"]  # base command
             if _pip_needs_system():
                 logger.debug("adding --system to work around pip3 defaulting to --user")
```

Comment:
I was tempted to remove this, but I waited to confirm it first. My understanding is that this verification was added to work around the stock Debian pip that defaults to `--user`, but since we're using the venv pip it seems that this is not needed anymore. I'll drop it.

Reply:
ack